web_widget_switchcase/static/src/js/web_widget_switchcase.js

```diff
@@ -0,0 +1,53 @@
+odoo.define('web_widget_switchcase', function (require) {
+    "use strict";
+
+    var core = require('web.core');
+    var formats = require('web.formats');
+    var form_widgets = require('web.form_widgets');
+
+    var CharSwitchCase = form_widgets.FieldChar.extend({

```

Comment:
Why do you make a new widget instead of extending (include) FieldChar with the functionality?